lib/make/makefile.epa-ng

```diff
@@ -13,22 +13,36 @@ SHELL := /bin/bash
 # Override in your Makefile by setting a parameter *after* the row that
 # includes this file, see documentation in doc/makefile.md.
 #
-__PGM___OPTS =
+EPA_NG_OPTS =
 
 # *** Internal ***
 
-# MAKECALL___PGM__ is a macro that defines what will be output to the .makecall
+# MAKECALL_EPA_NG is a macro that defines what will be output to the .makecall
 # file, the file that records versions, file stamps, parameters etc.
 #
 # *Don't redefine!*
-MAKECALL___PGM___VERSION     = echo "`date +"%Y%m%d %H:%M:%S"`: $@ was made with `__PROGRAM__ --version`" > $@.makecall
-MAKECALL___PGM___PARAMS      = echo "	Called with parameters: $(__PGM___OPTS)" >> $@.makecall
-MAKECALL___PGM___INFILES     = echo "	Input files: $^ (`ls -lL $^|tr '\n' ','`)" >> $@.makecall
-MAKECALL___PGM__             = $(MAKECALL___PGM___VERSION); $(MAKECALL___PGM___PARAMS); $(MAKECALL___PGM___INFILES)
+MAKECALL_EPA_NG_VERSION     = echo "`date +"%Y%m%d %H:%M:%S"`: $@ was made with `epa-ng --version | grep EPA-ng`" > $@.makecall
+MAKECALL_EPA_NG_PARAMS      = echo "    Called with parameters: $(__PGM___OPTS)" >> $@.makecall
+MAKECALL_EPA_NG_INFILES     = echo "    Input files: $^ (`ls -lL $^|tr '\n' ','`)" >> $@.makecall
+MAKECALL_EPA_NG             = $(MAKECALL_EPA_NG_VERSION); $(MAKECALL_EPA_NG_PARAMS); $(MAKECALL_EPA_NG_INFILES)
 
 # *** Targets ***
 
-%.suffix0: %.suffix1 %.suffix1 %.suffixn
-	@$(MAKECALL___PGM__)
-	__PGM_CALL__
-	@echo "`date +"%Y%m%d %H:%M:%S"`: DONE" >> $@.makecall
+# Target to rename the Reference Alignment, if it has been reduced by RAxML to match the epa_result target
+%.raxml.alnfaa: %.raxml.reduced.alnfaa
+        mv $(basename $(basename $@)).raxml.reduced.alnfaa $@
+
+# Here epa-ng takes as input: 
+# - ReferenceTree (rooted and safed as newick file)
+# - ReferenceAlignment (with the same id as the RAxML RT, raxml .phy file has been converted back to .alnfaa)
+# - QueryAlignment (created with hmmer)
+# 
+# epa-ng creates pre-named output files in the current directory:
+# - epa_info.log
+# - epa_results.jplace
+# so RT, RA and QS have to be in the same directory with no other RT, RA or QS files!
+
+epa_result.jplace: $(wildcard *.rooted.newick) $(wildcard *.raxml.alnfaa) $(wildcard *.hmmer.alnfaa)
```

Comment:
The `wildcard` calls here are expected to find just one file each. This is risky, as more than one file might match, and your numbering below is not correct. The problem you're trying to solve is, I believe, that the output file for the run is generic (`epa_result.jplace`). There are possibly two ways to solve this so you can turn it into a pattern target instead (i.e. with `%` in names). First, and easiest, there might be a flag with which you can specify the name of the output. If there isn't, you can rename the output file after a successful run, i.e. add a line `mv epa_result.jplace $@`. The disadvantage of the second approach is that it makes it impossible or unsafe to run multiple runs in the same directory.

Reply:
I can only specify an output dir (at least that's all I could find out) so the output will always be named `epa_result.jplace ` but I did not think about renaming the output itself (but I will also have to mv the logfile as it also has the same name every time). I will add that. Also: My query file was not named with the same prefix as the other files but that was more of a lazy mistake from me and I just have to adjust the filenames which is overall the better and smarter way.